Bring HikariCP in line with play-slick version and silence warnings

easel opened the pull request:
This PR brings the HikariCP version in line with the one used by play-slick, and sets a default for the validationTimeout value that will silence the log warning spam you'll see otherwise. The tests seem to pass, and this version of hikari-cp is going to see a lot of use as people migrate to play 2.4, so it seems reasonable to upgrade the dependency.

Happy to make any adjustments or add tests with a little instruction as to where.

typesafehub-validator commented:
Hi @easel,

Thank you for your contribution! We really value the time you've taken to put this together.

Before we proceed with reviewing this pull request, please sign the Typesafe Contributors License Agreement:

[http://www.typesafe.com/contribute/cla](http://www.typesafe.com/contribute/cla)

szeiger commented:
Thanks for the attempt but it seems to make things worse. However, according to https://github.com/brettwooldridge/HikariCP/blob/2.3.x/hikaricp-common/src/main/java/com/zaxxer/hikari/AbstractHikariConfig.java#L802 the requirement is not actually that the validationTimeout must be strictly less than the connectionTimeout (as indicated in the HikariCP docs) but only that it is less or equal, so using a default of 1000 should be fine and get rid of both the warning and the error.

szeiger commented on slick/src/main/scala/slick/jdbc/JdbcDataSource.scala:
```diff
@@ -150,6 +150,7 @@ object HikariCPJdbcDataSource extends JdbcDataSourceFactory {
 
     // Pool configuration
     hconf.setConnectionTimeout(c.getMillisecondsOr("connectionTimeout", 1000))
+    hconf.setValidationTimeout(c.getMillisecondsOr("validationTimeout", 500))
```
The new parameter should be document in the scaladoc for JdbcBackend.Database.forConfig

easel commented:
Ok, I've adjusted the default to 1000ms to match connection timeout and updated the docs. I also updated the docs for connection timeout to reflect the hikaricp docs for the minimum timeout. 

Now just to figure out why I didn't see any errors when I ran the tests locally =/

szeiger commented:
AFAIR the test error came from a doc test or streams test, not one of the standard TestKit tests, which don't use a connection pool. It would only show up in "testAll", not in "test".

easel commented:
Ok cool. I did figure out how to deploy a local snapshot and tested it without the workaround in my app successfully. Thanks for the help.
